src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
```

Comment:
This is hard to read. I would recommend splitting complex expressions to simpler ones using variables and functions.

Proposition:
```js
const women = people.filter((p) => {
  const hasChild = () => people.find((p1) => p1.mother === p.name);
  const isFemale = p.sex === "f";
  return isFemale && (!withChildren || hasChild());
});
  ```